### DISCUSSION TITLE
Bootstrap toast not showing in Livewire component

### DISCUSSION THREAD

**Discussion by adrianaditech** [livewire/livewire#8077] [1. Help] [answered]
In my livewire class I have:
`$this->dispatch('successfully-filed');`

In the component view I have:
`@script
        <script>
            $wire.on('successfully-filed', () => {
                $("#documentFiledToast").toast("show");
            });
        </script>
    @endscript`

and below that:
`<div class="toast" id="documentFiledToast">
        <div class="toast-header">
            <span class="text-success me-2"><i class="bi bi-check-circle-fill"></i></span>
            <strong class="me-auto">Successfully Filed</strong>
            <small class="text-body-secondary">just now</small>
            <button type="button" class="btn-close" data-bs-dismiss="toast" aria-label="Close"></button>
        </div>
        <div class="toast-body">
            Document was Successfully Filed
        </div>
    </div>`

No matter what I've tried, the toast never shows in the livewire component. I've even added vanilla js on page load event listener to show the toast but it doesn't do it (and no browser console output) in the component. Tried with and without jQuery.
I added the same js to my regular non-livewire blade views and it works fine.

How to fix it so the toast shows on that event?
I am firing other events in the same way ex. to show an offcanvas etc. and they all work, just not the toast.

**Comment by danie-ramdhani** [comment]
please read the bootstrap documentation how it works... (https://getbootstrap.com/docs/5.3/components/toasts/#live-example)

```js
@script
    <script>
        $wire.on('successfully-filed', () => {
            const toastBootstrap = bootstrap.Toast.getOrCreateInstance(document.getElementById("documentFiledToast"));
            toastBootstrap.show();
        });
    </script>
@endscript
```

**Reply by adrianaditech** [reply]
Thank you, but toast is not showing still. (My original code showed the toast properly when NOT in livewire component)

**Comment by Xinatorus** [comment]
I'm experiencing similar problems with tooltips. I thing it's because JS event are executed BEFORE DOM has changed. 
My current solution is wrapping called JS into setTimeout or jQuery $(function(){ ... })

**Comment by adrianaditech** [comment, marked as answer]
I troubleshooted that the toast actually was being shown with my original code, but it was in the very bottom corner of the page then was getting hidden immediately since the livewire component was being re-rendered.
To fix this I just put my toast into a 'toasts' partial and included it in my layout (not the livewire component) so it stays shown even when the component re-renders.

```
<div aria-live="polite" aria-atomic="true" class="position-relative">
    <div class="toast-container position-fixed top-0 end-0 p-3" style="z-index: 11">
        <div class="toast" id="documentFiledToast">
            <div class="toast-header">
                <span class="text-success me-2"><i class="bi bi-check-circle-fill"></i></span>
                <strong class="me-auto">Successfully Filed</strong>
                <small class="text-body-secondary">just now</small>
                <button type="button" class="btn-close" data-bs-dismiss="toast" aria-label="Close"></button>
            </div>
            <div class="toast-body">
                Document was Successfully Filed
            </div>
        </div>
    </div>
</div>

<script>
    document.addEventListener('livewire:init', () => {
        Livewire.on('successfully-filed', (event) => {
            console.log("successfully-filed");
            $("#documentFiledToast").toast("show");
        });
    });
</script>

```
